.github/workflows/cypress-workflow.yml

```diff
@@ -0,0 +1,129 @@
+name: E2E Cypress tests
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+jobs:
+  tests:
+    name: Run Cypress E2E tests
+    runs-on: ubuntu-latest
+    env:
+      # prevents extra Cypress installation progress messages
+      CI: 1
+      # avoid warnings like "tput: No value for $TERM and no -T specified"
+      TERM: xterm
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          # TODO: Parse this from security plugin
+          java-version: 13
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          path: security
+          repository: opendistro-for-elasticsearch/security
+      - name: Checkout Security Kibana plugin
+        uses: actions/checkout@v2
+        with:
+          path: security-kibana-plugin
+      - name: Get Kibana version
+        id: kibana_version
+        run: |
+          echo "::set-output name=kibana_version::$(node -p "(require('./security-kibana-plugin/package.json').kibana.version).match(/[.0-9]+/)[0]")"
+      - name: Get Opendistro version
+        id: opendistro_version
+        run: |
+          echo "::set-output name=opendistro_version::$(node -p "(require('./security-kibana-plugin/package.json').version).match(/[.0-9]+/)[0]")"
+      - name: Get Cypress version
+        id: cypress_version
+        run: |
+          echo "::set-output name=cypress_version::$(node -p "(require('./security-kibana-plugin/package.json').devDependencies.cypress).match(/[.0-9]+/)[0]")"
+      - name: Run elasticsearch with plugin
+        run: |
+          wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-${{ steps.kibana_version.outputs.kibana_version }}-darwin-x86_64.tar.gz
+          tar -xzf elasticsearch-oss-${{ steps.kibana_version.outputs.kibana_version }}-darwin-x86_64.tar.gz
+          cd elasticsearch-${{ steps.kibana_version.outputs.kibana_version }}/
+          bin/elasticsearch-plugin install -b https://d3g5vo6xdbdb9a.cloudfront.net/downloads/elasticsearch-plugins/opendistro-security/opendistro_security-${{ steps.opendistro_version.outputs.opendistro_version }}.zip
```

Comment:
Will this be always true ? What if feature is depending on backend code which is not released ?

Reply:
Managing cross pkg dependency is challenging. Currently for new releases, since backend plugin is released first, then Kibana plugin is released. So, for features which depend on backend code, the backend part can be committed into ES which will trigger a new build. Then Kibana side can get the latest build to run integ test against.